docs/config/index.md

```diff
@@ -27,7 +27,9 @@ You can also explicitly specify a config file to use with the `--config` CLI opt
 vite --config my-config.js
 ```
 
-Note that Vite will replace `__filename`, `__dirname`, and `import.meta.url`. Using these as variable names will result in an error:
+::: tip NOTE
+Vite will only replace `__filename`, `__dirname`, and `import.meta.url` in **CommonJS context** and **TS config file** (whether there is `type: "module"` or not). Using these as variable names will result in an error:
+:::
```

Comment:
Thanks for making the changes. After re-reading a few times, I think it sounds/looks a bit off 😬 I think the ending `:::` should always wrap pass the code block below, so that the `NOTE` block includes the code block.

I'm also thinking we can phrase the text as:

> Vite will replace `__filename`, `__dirname`, and `import.meta.url` in **CommonJS** and **TypeScript** config files. Using these as variable names will result in an error.